backend/app/api/src/helpers/UitpasTokenRepository.ts

```diff
@@ -0,0 +1,102 @@
+        const repo = UitpasTokenRepository.knownTokens.get(organizationId);
+        if (repo) {
+            return repo.getAccessToken(forceRefresh);
+        }
+        // query db
+        let uitpasToken = await UitpasToken.select().where('organizationId', organizationId).first(false);
+        if (!uitpasToken) {
+            // temporary solution, because platform client id and secret are not yet in the database
+            if (organizationId === null) {
+                if (!STAMHOOFD.UITPAS_API_CLIENT_ID || !STAMHOOFD.UITPAS_API_CLIENT_SECRET) {
+                    throw new SimpleError({
+                        code: 'uitpas_api_not_configured',
+                        message: 'UiTPAS API is not configured',
+                        human: $t('UiTPAS is niet volledig geconfigureerd. Neem contact op met de platformbeheerder.'),
+                    });
+                }
+                uitpasToken = new UitpasToken();
+                uitpasToken.clientId = STAMHOOFD.UITPAS_API_CLIENT_ID;
+                uitpasToken.clientSecret = STAMHOOFD.UITPAS_API_CLIENT_SECRET;
+                uitpasToken.organizationId = null; // null means platform
+            }
+            else {
+                throw new SimpleError({
+                    code: 'uitpas_token_not_found',
+                    message: organizationId ? `Uitpas token not found for clientId ${organizationId}` : 'Uitpas token not found for platform',
+                    human: $t(`De UiTPAS integratie is niet compleet, contacteer de beheerder.`),
+                });
+            }
+        }
+        const newRepo = new UitpasTokenRepository(uitpasToken);
+        this.knownTokens.set(organizationId, newRepo);
+        return newRepo.getAccessToken(true);
+    }
+
+    private async getAccessToken(forceRefresh: boolean = false): Promise<string> {
+        if (forceRefresh || !this.accessToken || this.expiresOn < new Date()) {
+            const url = 'https://account-test.uitid.be/realms/uitid/protocol/openid-connect/token';
+            const myHeaders = new Headers();
+            myHeaders.append('Content-Type', 'application/x-www-form-urlencoded');
+            const params = new URLSearchParams({
+                grant_type: 'client_credentials',
+                client_id: this.uitpasToken.clientId,
+                client_secret: this.uitpasToken.clientSecret,
+            });
+            const requestOptions: RequestInit = {
+                method: 'POST',
+                headers: myHeaders,
+                body: params.toString(),
+            };
+            const response = await fetch(url, requestOptions).catch((error) => {
+                // Handle network errors
+                throw new SimpleError({
+                    code: 'uitpas_unreachable',
+                    message: `Network error when fetching UiTPAS token for client id ${this.uitpasToken.clientId}: ${error.message}`,
```

Comment:
Zou de client id niet in de error message zetten, deze is niet echt private, maar toch ook niet iets wat je zichtbaar wilt maken